grow-with-freya/components/stories/story-book-reader.tsx

```diff
@@ -45,6 +47,62 @@ export function StoryBookReader({ story, onExit, onReadAnother, onBedtimeMusic }
   const [activeSubmenu, setActiveSubmenu] = useState<'main' | 'fontSize' | null>(null);
   const [readingMode, setReadingMode] = useState<'read' | 'record' | 'narrate'>('read');
 
+  // Voice recording state
+  const [isRecording, setIsRecording] = useState(false);
+  const [recordingDuration, setRecordingDuration] = useState(0);
+  const [currentVoiceOver, setCurrentVoiceOver] = useState<VoiceOver | null>(null);
+  const [showVoiceOverNameModal, setShowVoiceOverNameModal] = useState(false);
+  const [voiceOverName, setVoiceOverName] = useState('');
+  const [currentRecordingUri, setCurrentRecordingUri] = useState<string | null>(null);
+  const [tempRecordingUri, setTempRecordingUri] = useState<string | null>(null);
+  const [tempRecordingDuration, setTempRecordingDuration] = useState(0);
+  const [playbackSound, setPlaybackSound] = useState<Audio.Sound | null>(null);
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [availableVoiceOvers, setAvailableVoiceOvers] = useState<VoiceOver[]>([]);
+  const [showVoiceOverSelectModal, setShowVoiceOverSelectModal] = useState(false);
+  const [selectedVoiceOver, setSelectedVoiceOver] = useState<VoiceOver | null>(null);
+  const [isOverwriteSession, setIsOverwriteSession] = useState(false);
+  const [isNewVoiceOver, setIsNewVoiceOver] = useState(false);
+  const [narrationProgress, setNarrationProgress] = useState(0);
+  const [narrationDuration, setNarrationDuration] = useState(0);
+  const [shouldAutoStopRecording, setShouldAutoStopRecording] = useState(false);
+  const [shouldAutoAdvance, setShouldAutoAdvance] = useState(false);
+  const recordingTimerRef = useRef<ReturnType<typeof setInterval> | null>(null);
+  const textScrollViewRef = useRef<ScrollView>(null);
+
+  // Parents Only modal state
+  const [showParentsOnlyModal, setShowParentsOnlyModal] = useState(false);
+  const [parentsOnlyChallenge, setParentsOnlyChallenge] = useState<{ emoji: string; word: string }>({ emoji: '🐱', word: 'cat' });
+  const [parentsOnlyInput, setParentsOnlyInput] = useState('');
+  const parentsOnlyCallbackRef = useRef<(() => void) | null>(null);
+
+  const PARENT_CHALLENGES = [
+    { emoji: '🐱', word: 'cat' },
+    { emoji: '🦆', word: 'duck' },
+    { emoji: '🐕', word: 'dog' },
+    { emoji: '🐫', word: 'camel' },
+  ];
+
+  const showParentChallenge = (callback: () => void) => {
+    const randomChallenge = PARENT_CHALLENGES[Math.floor(Math.random() * PARENT_CHALLENGES.length)];
+    setParentsOnlyChallenge(randomChallenge);
+    setParentsOnlyInput('');
+    parentsOnlyCallbackRef.current = callback;
+    setShowParentsOnlyModal(true);
+  };
```

Comment:
The exact same parents only logic is duplicated in both this file and `main-menu.tsx`